internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -1689,6 +1689,12 @@ func (r *Reconciler) reconcilePostgresClusterDataSource(ctx context.Context,
 			client.ObjectKey{Name: sourceClusterName, Namespace: sourceClusterNamespace},
 			sourceCluster); err != nil {
 			if apierrors.IsNotFound(err) {
+				log.Error(
+					errors.New("DataSource refers to a non-existent PostgresCluster"),
```

Comment:
why are we creating a new error instead of using the error that occurred and having a msg on the next line that contains the text of this line?